functions

```diff
@@ -397,24 +386,47 @@ add_module() {
                 done
                 ;;
         esac
-    done < <(modinfo -b "$_optmoduleroot" -k "$KERNELVERSION" -0 "$target" 2>/dev/null)
+    done < <(modinfo -b "$_optmoduleroot" -k "$KERNELVERSION" -0 "${@%\?}" 2>/dev/null)
 
-    if (( !found )); then
-        (( ign_errors || _addedmodules["$target"] )) && return 0
-        error "module not found: \`%s'" "$target"
-        return 1
-    fi
+    for m in "${@//-/_}"; do
+        if [[ $m = *\? ]]; then
+            continue
+        fi
+
+        if (( _addedmodules[$m] )); then
+            continue
+        fi
+
+        warning "module not found: %s" "$m"
+    done
 
-    if (( ${#firmware[*]} )); then
-        add_firmware "${firmware[@]}" ||
-            warning 'Possibly missing firmware for module: %s' "$target"
+    if (( ! ${#_add_modules[@]} )); then
+        return
     fi
 
-    # handle module quirks
-    case $target in
+    _add_module "${_add_modules[@]}"
+}
+
+# add_module adds a kernel modules $@ and all it's dependencies and firmware
+# to the image. ?-suffix ignores errors.
```

Comment:
add_module adds $@ as kernel modules with dependencies and firmware to the image. A suffix of '?' causes errors to be ignored.